src/gatsby-node.js

```diff
@@ -7,22 +7,24 @@ import {
   checkForFaultyFields
 } from './faulty-keywords';
 
+const setHeaders = (origin, token) => {
+  let headers = {};
+  if (origin) {
+    headers = Object.assign(headers, {"Origin": origin})
+  }
+  if (token) {
+    headers = Object.assign(headers, {"Authorization": `Bearer ${token}`})
+  }
+  return Object.keys(headers).length === 0 ? {} : { headers };
```

Comment:
Using `isEmpty` from `ramda` would be more concise here. `isEmpty(headers) ? {} : { headers }`

Reply:
Ok great, thanks for the feedback.

I've renamed `setHeaders` to `createHeaders` and moved it to `util.js`
I've used Ramda's `isEmpty` to check the length of `headers`, and `merge` to join objects, and I've added tests for the 3 scenarios of this function.

My OS/Version is Mac High Sierra 10.13.6 (17G65)